packages/server-core/src/media/static-resource-vector/static-resource-vector.class.ts

```diff
@@ -0,0 +1,371 @@
+    }
+
+    const vectorData = {
+      ...data,
+      ...embeddings
+    }
+
+    if (existing.length > 0) {
+      // Update existing entry
+      return await this.patch(existing[0].id, vectorData)
+    } else {
+      // Create new entry
+      return await this.create(vectorData)
+    }
+  }
+
+  /**
+   * Delete vector entry by static resource ID
+   */
+  async deleteByStaticResourceId(staticResourceId: string): Promise<void> {
+    const existing = (await this.find({
+      query: { staticResourceId },
+      paginate: false
+    })) as StaticResourceVectorType[]
+
+    for (const entry of existing) {
+      await this.remove(entry.id)
+    }
+  }
+
+  /**
+   * Update vector entry when static resource is moved/renamed
+   */
+  async updateStaticResourceReference(oldStaticResourceId: string, newStaticResourceId: string): Promise<void> {
+    const existing = (await this.find({
+      query: { staticResourceId: oldStaticResourceId },
+      paginate: false
+    })) as StaticResourceVectorType[]
+
+    for (const entry of existing) {
+      await this.patch(entry.id, { staticResourceId: newStaticResourceId })
+    }
+  }
+
+  /**
+   * Override find method to handle semantic search queries
+   */
+  async find(params?: StaticResourceVectorParams): Promise<any> {
+    if (params?.query?.semanticSearch) {
+      const {
+        semanticSearch,
+        searchField = 'combined',
+        similarityThreshold = 0.7,
+        maxResults = 10,
+        ...otherQuery
+      } = params.query
+
+      // Perform semantic search
+      const results = await this.semanticSearch(semanticSearch, searchField as any, similarityThreshold, maxResults)
+
+      // If there are other query parameters, filter the results
+      if (Object.keys(otherQuery).length > 0) {
+        // Apply additional filters using the base find method
+        const filteredParams = { ...params, query: otherQuery }
+        const baseResults = await super.find(filteredParams)
+        const baseResultsArray = Array.isArray(baseResults) ? baseResults : baseResults.data || []
+
+        // Intersect semantic search results with filtered results
+        const filteredIds = new Set(baseResultsArray.map((r: any) => r.id))
+        return results.filter((r) => filteredIds.has(r.id))
+      }
+
+      return results
+    }
+
+    return super.find(params)
+  }
+
+  /**
+   * Sync static resource data to vector database
+   */
+  async syncStaticResource(staticResource: any): Promise<void> {
+    try {
+      // Extract searchable fields from static resource
+      const vectorData = {
+        staticResourceId: staticResource.id,
+        caption: staticResource.description || staticResource.name || '',
+        material: this.extractMaterial(staticResource),
+        style: this.extractStyle(staticResource),
+        type: staticResource.type || staticResource.mimeType || '',
+        color: this.extractColor(staticResource)
```

Comment:
comment these out, we don't need to extract these fields. Let's wait for the CV processor to produce these fields for us